service.mailnotifiers/notifier.py

```diff
@@ -0,0 +1,256 @@
+font = Addon.getSetting('font')
+color = Addon.getSetting('color')
+# Alternate display ??
+ALT = Addon.getSetting('alt')
+# Display in the skin
+SKIN = Addon.getSetting('skin')
+# Display on multilines
+if (Addon.getSetting('multilines').lower == 'true'):
+    SEP = "\n"
+else:
+    SEP = '| '
+# Control ID in window Home.xml
+MsgBox = None
+MsgBoxId = None
+
+start_time = 0
+# Flag for add or not control in HOME
+re_added_control = False
+
+if __name__ == '__main__':
+    monitor = xbmc.Monitor()
+ # Verify if kodi work
+    while not monitor.abortRequested():
+        # Sleep/wait for abort for 10 seconds
+        # Wait before get mails
+        intervalle = int(float(Addon.getSetting('time')) * 60.0)
+        if monitor.waitForAbort(intervalle):
+            # Abort was requested while waiting. We should exit
+            break
+
+        if start_time and (time.time() - start_time) < intervalle:
+            monitor.waitForAbort(.5)
+            SHOW_UPDATE     = Addon.getSetting('show_update') == "true"
+            # if control exist
+            if MsgBox:
+                # optional show time before next update
+                try:
+                    # If SHOW_UPDATE true
+                    if SHOW_UPDATE:
+                        up = int(intervalle) - (time.time() - start_time)
+                        locstr = Addon.getLocalizedString(615)  # Update in %i second
+                        xbmc.log(("[%s] : MSG up = %s " % (scriptname, msg),DEBUG_LOG))
+                        label = "%s[CR], %s : %s" % (msg, locstr, up)
+                        debug_string = "Msg = %s, Update = %s" % (msg, up)
+                        xbmc.log(("[%s] : label = %s " % debug_string),DEBUG_LOG)
+                    else:  # Need to refresh display
+                        xbmc.log(("[%s] : MSG = %s " % (scriptname, msg),DEBUG_LOG))
+                        label = '%s' % msg
+                    if (SKIN == "false"):
+                        MsgBox.setLabel(msg)
+                        xbmc.log(("[%s] : setlabel : %s" % (scriptname, msg),DEBUG_LOG))
+                    else:
+                        MsgBox.setLabel('')
+                        debug('Clean label')
+                except Exception, e:
+                    print str(e)
+            # End of if MsgBox
+            HomeNotVisible = xbmc.getCondVisibility("!Window.IsVisible(10000)")
+            if HomeNotVisible:
+                # oop! not on the home
+                re_added_control = True
+            # elif re_added_control and not HomeNotVisible:
+            else:
+                # Try to get getcontrol if not exist make a new one
+                try:
+                    MsgBox = homeWin.getControl(MsgBoxId)
+                except:
+                    MsgBox = xbmcgui.ControlLabel(x, y, width, height, '', font, color)
+                # add control label and set default label
+                try:
+                    homeWin.addControl(MsgBox)
```

Comment:
adding a control on-the-fly to home screen feels rather hacky.. not possible to just use a Notification?